index.js

```diff
@@ -11,17 +11,16 @@ var resolve = require('resolve')
 var getPackageInfo = require('get-package-info')
 var common = require('./common')
 
-var supportedArchs = {
-  ia32: 1,
-  x64: 1
-}
-
-var supportedPlatforms = {
-  // Maps to module ID for each platform (lazy-required if used)
-  darwin: './mac',
-  linux: './linux',
-  win32: './win32'
-}
+var supportedArchs = common.archs.reduce(function (o, v) {
+  o[v] = 1
+  return o
+}, {})
+
+// Maps to module ID for each platform (lazy-required if used)
+var supportedPlatforms = common.platforms.reduce(function (o, v) {
+  o[v] = './' + (v === 'darwin' ? 'mac' : v)
+  return o
+}, {})
```

Comment:
I also prefer to use `for of` instead of magic `function style`. But I see `forEach` in existing code.

Do you want to convert it to

```js
var supportedPlatforms = {}
for (var platform in common.platforms) {
  supportedPlatforms[platform] = './' + (platform === 'darwin' ? 'mac' : platform)
}
```
?